oauth2/fosite_store_sql.go

```diff
@@ -289,7 +290,14 @@ func (s *FositeSQLStore) createSession(ctx context.Context, signature string, re
 		":"+strings.Join(sqlParams, ", :"),
 	)
 	if _, err := db.NamedExecContext(ctx, query, data); err != nil {
-		return sqlcon.HandleError(err)
+		err = sqlcon.HandleError(err)
+		switch errCause := errors.Cause(err).(type) {
+		case *herodot.DefaultError:
+			if errCause == sqlcon.ErrConcurrentUpdate {
+				return errors.Wrap(fosite.ErrSerializationFailure, err.Error())
+			}
+		}
+		return err
```

Comment:
The reasoning here is to avoid `fosite` having a dependency on `sqlcon`. So the error is mapped from `sqlcon.ErrConcurrentUpdate` to `fosite.ErrSerializationFailure` from which `fosite` will be able to handle and return an appropriate RFC 6749 error.